server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -103,23 +106,27 @@ public void start()
         return;
       }
 
-      this.exec = Execs.scheduledSingleThreaded("DatabaseSegmentManager-Exec--%d");
+      exec = MoreExecutors.listeningDecorator(Execs.scheduledSingleThreaded("DatabaseSegmentManager-Exec--%d"));
 
       final Duration delay = config.get().getPollDuration().toStandardDuration();
-      ScheduledExecutors.scheduleWithFixedDelay(
-          exec,
-          new Duration(0),
-          delay,
+      future = exec.scheduleWithFixedDelay(
           new Runnable()
           {
             @Override
             public void run()
             {
-              poll();
+              try {
+                poll();
+              }
+              catch (Exception e) {
+                log.error(e, "uncaught exception in segment manager polling thread");
```

Comment:
I just realized this error will never get logged, `SQLMetadataSegmentManager.poll()` already logs and swallows all exceptions